addons/mass_mailing/static/src/js/unsubscribe.js

```diff
@@ -97,94 +102,110 @@ odoo.define('mass_mailing.unsubscribe', function (require) {
     //  ==================
     $('#button_add_blacklist').click(function (e) {
         e.preventDefault();
-
-        ajax.jsonRpc('/mailing/blacklist/add', 'call', {'email': email, 'mailing_id': mailing_id, 'res_id': res_id, 'token': token})
-            .then(function (result) {
-                if (result == 'unauthorized'){
-                    $('#subscription_info').html(_t('You are not authorized to do this!'));
-                    $('#info_state').removeClass('alert-success').removeClass('alert-info').removeClass('alert-error').addClass('alert-warning');
-                }
-                else
-                {
-                    if (result) {
-                        $('#subscription_info').html(_t('You have been successfully <strong>added to our blacklist</strong>. '
-                               + 'You will not be contacted anymore by our services.'));
-                        $('#info_state').removeClass('alert-warning').removeClass('alert-info').removeClass('alert-error').addClass('alert-success');
-                        toggle_opt_out_section(false);
+        if(!$('#button_add_blacklist').hasClass('disabled')){
```

Comment:
You cannot click on a disabled button anyway, right ?
and Oh gosh, I did not remember I wrote such a crap.. this should be refactored in a widget, use local selector, declare events using proper 'widget way', etc.. (just a note for me :smile: )